lib/core/src/lib/testing/unit-testing-utils.ts

```diff
@@ -0,0 +1,465 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { HarnessLoader, TestElement, TestKey } from '@angular/cdk/testing';
+import { DebugElement } from '@angular/core';
+import { By } from '@angular/platform-browser';
+import { MatSelectHarness } from '@angular/material/select/testing';
+import { MatOptionHarness } from '@angular/material/core/testing';
+import { MatChipGridHarness, MatChipHarness, MatChipListboxHarness } from '@angular/material/chips/testing';
+import { MatButtonHarness } from '@angular/material/button/testing';
+import { MatIconHarness } from '@angular/material/icon/testing';
+import { MatCheckboxHarness } from '@angular/material/checkbox/testing';
+import { MatErrorHarness, MatFormFieldHarness } from '@angular/material/form-field/testing';
+import { MatInputHarness } from '@angular/material/input/testing';
+import { MatAutocompleteHarness } from '@angular/material/autocomplete/testing';
+import { ComponentFixture } from '@angular/core/testing';
+import { MatTabGroupHarness, MatTabHarness } from '@angular/material/tabs/testing';
+import { MatToolbarHarness } from '@angular/material/toolbar/testing';
+import { MatSnackBarHarness } from '@angular/material/snack-bar/testing';
+import { MatProgressBarHarness } from '@angular/material/progress-bar/testing';
+
+export class UnitTestingUtils {
+    static getByCSS(debugElement: DebugElement, selector: string): DebugElement {
+        return debugElement.query(By.css(selector));
+    }
+
+    static getAllByCSS(debugElement: DebugElement, selector: string): DebugElement[] {
+        return debugElement.queryAll(By.css(selector));
+    }
+
+    static getInnerTextByCSS(debugElement: DebugElement, selector: string): string {
+        return UnitTestingUtils.getByCSS(debugElement, selector).nativeElement.innerText;
+    }
+
+    static getByDataAutomationId(debugElement: DebugElement, dataAutomationId: string): DebugElement {
+        return UnitTestingUtils.getByCSS(debugElement, `[data-automation-id="${dataAutomationId}"]`);
+    }
+
+    static getByDataAutomationClass(debugElement: DebugElement, dataAutomationClass: string): DebugElement {
+        return UnitTestingUtils.getByCSS(debugElement, `[data-automation-class="${dataAutomationClass}"]`);
+    }
+
+    static getAllByDataAutomationId(debugElement: DebugElement, dataAutomationId: string): DebugElement[] {
+        return UnitTestingUtils.getAllByCSS(debugElement, `[data-automation-id="${dataAutomationId}"]`);
+    }
+
+    static getInnerTextByDataAutomationId(debugElement: DebugElement, dataAutomationId: string): string {
+        return UnitTestingUtils.getByDataAutomationId(debugElement, dataAutomationId).nativeElement.innerText;
+    }
+
+    static getByDirective(debugElement: DebugElement, directive: any): DebugElement {
```

Comment:
instead of any for directive, could you please use ``Type<any>``? It will be a bit safer because it won't allow some values like for example strings.

Reply:
Good idea, done